pkg/controller/controller_instance.go

```diff
@@ -758,6 +754,13 @@ func (c *controller) pollServiceInstance(instance *v1beta1.ServiceInstance) erro
 	}
 }
 
+// clearServiceInstanceAsyncOperation will reset the given instance's async
+// operation-related status fields.
+func clearServiceInstanceAsyncOperation(instance *v1beta1.ServiceInstance) {
```

Comment:
Given that this is a new method that is used only for deprovisioning, maybe it should be named differently (with more specifics)?
We have another `clearServiceInstanceCurrentOperation` method with a similar name and code, but has a different purpose.
It should at least explicitly mention that we reset only the temporary async status, while the `instance.Status.CurrentOperation` itself is being kept and the operation will be retried.

Reply:
Yeah, that's an unfortunate consequence of the fact that OSB considers an "operation" different  from what Service Catalog does. I hesitate to name it something more specific to deprovisioning, as it is in function quite general and could be used elsewhere. Maybe `clearServiceInstanceAsyncOsbOperation`? I can certainly add clarification in the comment, though.